### PR TITLE
Review DB semantic conventions

### PR DESCRIPTION
**Pull request by iNikem**

### CODE REVIEW

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java [hunk 1]
```diff
@@ -111,32 +112,43 @@ protected void onError(Span span, Throwable throwable) {
...
+  }
 
-  // TODO make abstract after implementing in all subclasses
   protected String dbConnectionString(CONNECTION connection) {
     return null;
   }
 
   protected abstract InetSocketAddress peerAddress(CONNECTION connection);
+
+  // TODO: "When it's impossible to get any meaningful representation of the span name, it can be
```
Comment (anuraaga) [inline instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java:141]: I realize this is just copied, but maybe remove `TODO`, it seems to be an implementation of the spec with no todos from what I can tell.

**Review by anuraaga — APPROVED** [review]
No decorator to tracer migration this time, what happened? 😿 😄

File: instrumentation/cassandra/cassandra-3.0/src/main/java/io/opentelemetry/instrumentation/auto/cassandra/v3_0/CassandraDatabaseClientTracer.java [hunk 2]
```diff
@@ -34,23 +36,24 @@ protected String getInstrumentationName() {
   }
 
   @Override
-  protected String normalizeQuery(String query) {
+  protected @NonNull String normalizeQuery(String query) {
```
[the remark this answers is not in the document]
Comment (iNikem) [reply instrumentation/cassandra/cassandra-3.0/src/main/java/io/opentelemetry/instrumentation/auto/cassandra/v3_0/CassandraDatabaseClientTracer.java:39]: I added it here to be sure that no subclass return `null`.

**Comment by iNikem** [comment]
> No decorator to tracer migration this time, what happened? 😿 😄

That was done in #1236 :)

File: instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java (hunk 1 above)
Comment (iNikem) [reply instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/DatabaseClientTracer.java:141]: O yes, I have fixed this TODO in this PR. Will remove

File: instrumentation/cassandra/cassandra-3.0/src/main/java/io/opentelemetry/instrumentation/auto/cassandra/v3_0/CassandraDatabaseClientTracer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (trask) [reply instrumentation/cassandra/cassandra-3.0/src/main/java/io/opentelemetry/instrumentation/auto/cassandra/v3_0/CassandraDatabaseClientTracer.java:39]: If a subclass returns null, would it lead to a NullPointerException? If not, I'd suggest we don't need the `@NonNull` annotation on the return type

[the remark this answers is not in the document]
Comment (iNikem) [reply instrumentation/cassandra/cassandra-3.0/src/main/java/io/opentelemetry/instrumentation/auto/cassandra/v3_0/CassandraDatabaseClientTracer.java]: Removed

**Review by trask — APPROVED** [review]

**Pull request merged**